server/controllers/pages.js

```diff
@@ -16,7 +16,27 @@ const entryHelper = require('../helpers/entry')
  * Edit document in Markdown
  */
 router.get('/edit/*', (req, res, next) => {
-  if (!res.locals.rights.write) {
+  var xPaths = ['news', 'general'] // the final word in the url For.e.g : news in /edit/news, general in /edit/general 
+
+  var checkEditAvailable = function (xPaths) {
+    let paths = req.path.split('/')
```

Comment:
Change to `let page_path = req.path.split('/edit', 2)[1];`.

That way, you directly have the path you want to compare to.